Wikipedia/Code/TalkPageViewController.swift

```diff
@@ -303,6 +298,91 @@ class TalkPageViewController: ViewController {
             WMFAlertManager.sharedInstance.showBottomAlertWithMessage(title, subtitle: nil, image: image, dismissPreviousAlerts: true)
         }
     }
+    
+    // MARK: - Scrolling Helpers
+    
+    private func scrollToLastTopic() {
+        if viewModel.topics.count > 0 {
+            let indexPath = IndexPath.init(item: viewModel.topics.count - 1, section: 0)
+            talkPageView.collectionView.scrollToItem(at: indexPath, at: .top, animated: true)
+        }
+    }
+    
+    private func scrollToNewComment(oldCellViewModel: TalkPageCellViewModel?, oldCommentViewModels: [TalkPageCellCommentViewModel]?) {
+        
+        guard let oldCellViewModel = oldCellViewModel,
+        let oldCommentViewModels = oldCommentViewModels else {
+            return
+        }
+        
+        let newCellViewModel = viewModel.topics.first { $0 == oldCellViewModel }
+        
+        guard let newCommentViewModels = newCellViewModel?.allCommentViewModels else {
+            return
+        }
+        
+        if let newCommentViewModel = newestComment(from: oldCommentViewModels, to: newCommentViewModels) {
+            scrollToComment(commentViewModel: newCommentViewModel)
+        }
+    }
+    
+    private func newestComment(from oldCommentViewModels: [TalkPageCellCommentViewModel], to newCommentViewModels: [TalkPageCellCommentViewModel]) -> TalkPageCellCommentViewModel? {
+        
+        let oldCommentViewModelsSet = Set(oldCommentViewModels)
+        let newCommentViewModelsSet = Set(newCommentViewModels)
+        
+        let newComments = newCommentViewModelsSet.subtracting(oldCommentViewModelsSet)
+        
+        // MAYBETODO: Sort by date here?
+        
+        guard let newComment = newComments.first else {
+            return nil
+        }
+        
+        return newComment
+    }
+    
+    private func scrollToComment(commentViewModel: TalkPageCellCommentViewModel) {
```

Comment:
My hope is that this is will be reusable-enough for navigating through our eventual find in page results.

Reply:
👍🏽 I've been playing with a few ideas to simplify this, but so far your solution seems to be working quite nicely for me. I anticipate with find in page we'll probably have to parameterize out the animation properties.